File: premiumhub-web/src/types/order.ts
export interface OrderProduct {
  id?: string
  name: string
  icon: string
  color?: string
}

export interface OrderPrice {
  id?: string
  product_id?: string
  duration: number
  account_type: string
  price: number
  is_active?: boolean
}

export interface OrderStock {
  id?: string
  product_id?: string
  account_type?: string
  email: string
  password?: string
  profile_name?: string
  status?: string
  used_by?: string
  used_at?: string | null
  expires_at?: string | null
}

export interface OrderUser {
  id: string
  name: string
  email: string
  phone?: string
  role?: string
  is_active?: boolean
}

export interface Order {
  id: string
  user_id: string
  stock_id?: string | null
  price_id?: string
  product?: OrderProduct
  user?: OrderUser
  price: OrderPrice
  total_price: number
  payment_method?: string
  payment_status: 'pending' | 'paid' | 'failed' | 'expired'
  order_status: 'pending' | 'active' | 'completed' | 'failed'
  stock?: OrderStock
  gateway_order_id?: string
  payment_payload?: string
  paid_at: string | null
  expires_at?: string | null
  created_at: string
  updated_at?: string
}

export interface Claim {
  id: string
  user_id: string
  order_id: string
  reason: 'login' | 'password' | 'kicked' | 'profile' | 'quality' | 'other' | string
  description: string
  screenshot_url?: string
  status: 'pending' | 'approved' | 'rejected'
  admin_note?: string
  new_stock_id?: string | null
  resolved_at?: string | null
  user?: OrderUser
  order?: Order
  created_at: string
  updated_at?: string
}

File: premiumhub-api/internal/model/order.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Order struct {
	ID             uuid.UUID    `gorm:"type:uuid;primaryKey" json:"id"`
	UserID         uuid.UUID    `gorm:"type:uuid;not null" json:"user_id"`
	StockID        *uuid.UUID   `gorm:"type:uuid" json:"stock_id"`
	PriceID        uuid.UUID    `gorm:"type:uuid;not null" json:"price_id"`
	TotalPrice     int64        `gorm:"not null" json:"total_price"`
	PaymentMethod  string       `gorm:"size:50" json:"payment_method"`
	PaymentStatus  string       `gorm:"size:20;default:pending" json:"payment_status"`
	OrderStatus    string       `gorm:"size:20;default:pending" json:"order_status"`
	GatewayOrderID string       `gorm:"size:120;index" json:"gateway_order_id"`
	PaymentPayload string       `gorm:"type:text" json:"payment_payload,omitempty"`
	PaidAt         *time.Time   `json:"paid_at"`
	ExpiresAt      *time.Time   `json:"expires_at"`
	User           User         `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Stock          *Stock       `gorm:"foreignKey:StockID" json:"stock,omitempty"`
	Price          ProductPrice `gorm:"foreignKey:PriceID" json:"price,omitempty"`
	CreatedAt      time.Time    `json:"created_at"`
	UpdatedAt      time.Time    `json:"updated_at"`
}

func (o *Order) BeforeCreate(_ *gorm.DB) error {
	if o.ID == uuid.Nil {
		o.ID = uuid.New()
	}
	return nil
}
